CRM/Upgrade/Incremental/php/FiveFiftyNine.php

```diff
@@ -30,6 +30,17 @@ class CRM_Upgrade_Incremental_php_FiveFiftyNine extends CRM_Upgrade_Incremental_
   public function upgrade_5_59_alpha1($rev): void {
     $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
     $this->addTask('Drop column civicrm_custom_field.mask', 'dropColumn', 'civicrm_custom_field', 'mask');
+    $this->addTask('Add scheduled_reminder_smarty setting', 'addScheduledReminderSmartySetting');
+  }
+
+  public function setPostUpgradeMessage(&$postUpgradeMessage, $rev): void {
+    if ($rev === '5.59.alpha1') {
+      $postUpgradeMessage .= 'You can now choose whether to use Smarty in Scheduled Reminders via the scheduled_reminder_smarty setting. We have enabled the setting during this upgrade to preserve the existing behaviour but you may wish to disable it now. Note: the setting is disabled by default in new installations. More information here: https://lab.civicrm.org/dev/core/-/issues/4100.';
+    }
+  }
+
+  public function addScheduledReminderSmartySetting() {
+    Civi::settings()->set('scheduled_reminder_smarty', TRUE);
```

Comment:
Tasks need to return TRUE (or FALSE, but it's usually true that they return TRUE).

Reply:
Wouldn't this be better handled as a default than an upgrade task?

Reply:
Yes - I agree with @agileware-fj

Reply:
Thanks for the feedback @agileware-fj - it is now a setting registered in Mailing.setting.php

Reply:
However, I have kept this upgrade task in because we want to change the default on new installations (because it is a more sensibe default) but preserve the existing behaviour for new users (to minimise disruption).

Reply:
@michaelmcandrew can you shift this upgrade stuff to FiveSixty.php as 5.59 is now an RC

Reply:
Sure - done.